tests/framework/randomness/test_crn.py

```diff
@@ -0,0 +1,301 @@
+"""
+Integration tests primarily meant to test the CRN guarantees for the Randomness system.
+
+"""
+from itertools import cycle
+from typing import Iterator
+
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+from vivarium.framework.randomness.index_map import IndexMap
+from vivarium.framework.randomness.stream import RandomnessStream
+from vivarium.interface import InteractiveContext
+
+
+@pytest.mark.parametrize("for_initialization", [True, False])
+def test_basic_repeatability(for_initialization):
+    test_idx = pd.Index(range(100))
+    index_map = IndexMap()
+
+    stream_args = {
+        "key": "test",
+        "clock": lambda: pd.Timestamp("2020-01-01"),
+        "seed": "abc",
+        "index_map": index_map,
+        "for_initialization": for_initialization,
+    }
+
+    stream_base = RandomnessStream(**stream_args)
+    draw_base = stream_base.get_draw(test_idx)
+
+    for arg_permutation in [
```

Comment:
Thoughts on this test parameterization being internal to the test rather than in the `@pytest.mark.parameterize` annotation?

Reply:
:shrug: